12-offer-encoding.md

```diff
@@ -0,0 +1,1057 @@
+    - MUST reject the invoice if `blinded_payinfo` is not present.
+    - MUST reject the invoice if `blinded_payinfo` does not contain exactly as many `payinfo` as total `onionmsg_path` in `blinded_path`.
+  - SHOULD confirm authorization if `msat` is not within the amount range authorized.
+  - if the invoice is a reply to an `invoice_request`:
+     - MUST reject the invoice unless `offer_id` is equal to the id of the offer.
+     - MUST reject the invoice unless `node_id` is equal to the offer.
+     - MUST reject the invoice unless the following fields are equal or unset
+       exactly as they are in the `invoice_request:`
+       - `quantity`
+       - `recurrence_counter`
+       - `recurrence_start`
+       - `payer_key`
+       - `payer_info`
+    - MUST reject the invoice if `payer_note` is set, and was unset or not equal to the field in the `invoice_request`.
+    - SHOULD confirm authorization if the `description` does not exactly
+      match the `offer`
+      - MAY highlight if `description` has simply had a change appended.
+    - SHOULD confirm authorization if `issuer` does not exactly
+      match the `offer`.
+  - otherwise if `offer_id` is set:
+    - MUST reject the invoice if the `offer_id` does not refer an unexpired offer with `send_invoice`
+    - MUST reject the invoice unless the following fields are equal or unset
+      exactly as they are in the `offer`:
+      - `refund_for`
+      - `description`
+    - if the offer had a `quantity_min` or `quantity_max` field:
+      - MUST reject the invoice if there is no `quantity` field.
+      - MUST reject the invoice if there is `quantity` is not within that (inclusive) range.
+    - otherwise:
+      - MUST reject the invoice if there is a `quantity` field.
+  - if the offer contained `recurrence`:
+    - MUST reject the invoice if `recurrence_basetime` is not set.
+  - if the offer contained `refund_for`:
+      - MUST reject the invoice if `payer_key` does not match the invoice whose `payment_hash` is equal to `refund_for` `refunded_payment_hash`
+      - MUST reject the invoice if `refund_signature` is not set.
+      - MUST reject the invoice if `refund_signature` is not a valid signature using `payer_key` as described in [Signature Calculation](#signature-calculation).
+  - for the bitcoin chain, if the invoice specifies `fallbacks`:
+    - MUST ignore any `fallback_address` for which `version` is greater than 16.
+    - MUST ignore any `fallback_address` for which `address` is less than 2 or greater than 40 bytes.
+    - MUST ignore any `fallback_address` for which `address` does not meet known requirements for the given `version`
+
+## Rationale
+
+Because the messaging layer is unreliable, it's quite possible to
+receive multiple requests for the same offer.  As it's the caller's
+responsibility not to reuse `payer_key` except for recurring invoices,
+the writer doesn't have to check all the fields are duplicates before
+simply returning a previous invoice.
```

Comment:
I understand the rationale but the writer should at least check that the invoice hasn't expired.